src/pages/api/products/[id].tsx

```diff
@@ -0,0 +1,31 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+
+const productHandler = (req: NextApiRequest, res: NextApiResponse) => {
+  const { method, query: { id } } = req
+
+  const handlePatch = () => {
+    return {
+      message: `This is a PATCH request, it should patch id: ${id}. Currently in progress...`,
+      id: id
+    }
+  }
+
+  const handleDelete = () => {
+    return {
+      message: `This is a DELETE request, it should DELETE id: ${id}. Currently in progress...`,
+      id: id
+    }
+  }
+
+  switch (method) {
+    case 'PATCH':
+      return res.json(handlePatch())
+    case 'DELETE':
+      return res.json(handleDelete())
+    default:
+      res.setHeader('Allow', ['PATCH', 'DELETE'])
+      res.status(405).end(`Method ${method} Not Allowed`)
+  }
+}
```

Comment:
I have 2 concerns
1. I don't think we need to declare a function just for returning an object. If you are planing to implement it later, then  please add `TODO` for others to understand.
2. If you agree with my previous concern, please specify its type also. 
Ex. `Message` in my suggestion is the interface name

PS. my variable names are just the example, you can change it to whatever you want.

```suggestion
  const patchMessage: Message = {
    message: `This is a PATCH request, it should patch id: ${id}. Currently in progress...`,
    id: id,
  }

  const deleteMessage: Message = {
    message: `This is a DELETE request, it should DELETE id: ${id}. Currently in progress...`,
    id: id,
  }

  switch (method) {
    case 'PATCH':
      return res.json(patchMessage)
    case 'DELETE':
      return res.json(deleteMessage)
    default:
      res.setHeader('Allow', ['PATCH', 'DELETE'])
      res.status(405).end(`Method ${method} Not Allowed`)
  }
}
```

Reply:
Added `TODO`.